lib/Backend/Pass.hs

```diff
@@ -0,0 +1,170 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Backend.Pass
+  ( PassBackend ) where
+import Backend
+import BackendName
+import Coffer.Path
+import Coffer.Path qualified as P
+import Control.Exception (IOException)
+import Control.Lens
+import Data.ByteString.Lazy (ByteString)
+import Data.ByteString.Lazy qualified as BS
+import Data.Maybe
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Text.Encoding (encodeUtf8)
+import Data.Text.Encoding qualified as T
+import Effect.Fs
+import Entry (Entry)
+import Entry qualified as E
+import Entry.Pass
+import Error
+import Fmt (build, fmt)
+import Polysemy
+import Polysemy.Error
+import System.Directory qualified as D
+import System.FilePath (makeRelative)
+import System.IO.Error (isDoesNotExistError)
+import System.Process.Typed
+import Toml (TomlCodec)
+import Toml qualified
+
+data PassBackend =
+  PassBackend
+  { pbName :: BackendName
+  , pbStoreDir :: FilePath
+  , pbPassExe :: Maybe FilePath
+  }
+  deriving stock (Show)
+
+passCodec :: TomlCodec PassBackend
+passCodec =
+  PassBackend
+  <$> backendNameCodec "name" Toml..= pbName
+  <*> Toml.string "store_dir" Toml..= pbStoreDir
+  <*> Toml.dimatch fPathToT tToFPath (Toml.text "pass_exe") Toml..= pbPassExe
+  where tToFPath = Just . T.unpack
+        fPathToT :: Maybe String -> Maybe Text
+        fPathToT a = a <&> T.pack
+
+
```

Comment:
According to [styleguide](https://github.com/serokell/style/blob/master/haskell.md#blank-lines) we must add one blank line between top-level definitions.